ApplicationLibCode/Commands/OsduImportCommands/RiuWellImportWizard.cpp

```diff
@@ -0,0 +1,502 @@
+//
+//  Copyright (C) 2011-2012 Statoil ASA, Ceetron AS
+//
+//  ResInsight is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+//
+//  ResInsight is distributed in the hope that it will be useful, but WITHOUT ANY
+//  WARRANTY; without even the implied warranty of MERCHANTABILITY or
+//  FITNESS FOR A PARTICULAR PURPOSE.
+//
+//  See the GNU General Public License at <http://www.gnu.org/licenses/gpl.html>
+//  for more details.
+//
+/////////////////////////////////////////////////////////////////////////////////
+
+#include "RiuWellImportWizard.h"
+
+#include "RiaFeatureCommandContext.h"
+#include "RiaOsduConnector.h"
+
+#include "RimWellPathImport.h"
+
+#include "cafCmdFeatureMenuBuilder.h"
+#include "cafPdmUiListView.h"
+#include "cafPdmUiPropertyView.h"
+#include "cafPdmUiTreeAttributes.h"
+#include "cafPdmUiTreeView.h"
+#include "cafPdmUiTreeViewEditor.h"
+#include "cafUtils.h"
+
+#include <QAbstractTableModel>
+#include <QObject>
+#include <QString>
+
+#include <QtNetwork>
+#include <QtWidgets>
+
+#include <algorithm>
+#include <optional>
+#include <vector>
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+RiuWellImportWizard::RiuWellImportWizard( const QString&     downloadFolder,
+                                          RiaOsduConnector*  osduConnector,
+                                          RimWellPathImport* wellPathImportObject,
+                                          QWidget*           parent /*= 0*/ )
+    : QWizard( parent )
+{
+    m_wellPathImportObject = wellPathImportObject;
+
+    m_osduConnector     = osduConnector;
+    m_destinationFolder = downloadFolder;
+
+    m_firstTimeRequestingAuthentication = true;
+
+    addPage( new AuthenticationPage( m_osduConnector, this ) );
+    addPage( new FieldSelectionPage( m_wellPathImportObject, m_osduConnector, this ) );
+    addPage( new WellSelectionPage( m_wellPathImportObject, m_osduConnector, this ) );
+    addPage( new WellSummaryPage( m_wellPathImportObject, m_osduConnector, this ) );
+}
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+RiuWellImportWizard::~RiuWellImportWizard()
+{
+}
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+void RiuWellImportWizard::downloadFields()
+{
+    m_osduConnector->requestFieldsByName( "AZERI" );
```

Comment:
Add TODO label to clearly mark temporary code

Reply:
Fixed.